HealthVault/Classes/Clients/MHVPersonClient.m

```diff
@@ -53,13 +56,70 @@ - (instancetype)initWithConnection:(id<MHVConnectionProtocol>)connection
 
 - (void)getApplicationSettingsWithCompletion:(void(^)(MHVApplicationSettings *_Nullable settings, NSError *_Nullable error))completion
 {
+    MHVASSERT_PARAMETER(completion);
+    
+    if (!completion)
+    {
+        return;
+    }
     
+    MHVMethod *method = [MHVMethod getApplicationSettings];
+    
+    [self.connection executeHttpServiceOperation:method
+                                      completion:^(MHVServiceResponse * _Nullable response, NSError * _Nullable error)
+     {
+         if (error)
+         {
+             completion(nil, error);
+             
+             return;
+         }
+         
+         MHVApplicationSettings *applicationSettings;
+         
+         XReader *reader = [[XReader alloc] initFromString:response.infoXml];
+         
+         if ([reader readStartElementWithName:@"info"])
+         {
+             applicationSettings = (MHVApplicationSettings *)[reader readElement:@"app-settings" asClass:[MHVApplicationSettings class]];
+         }
+         
+         if (!applicationSettings)
+         {
+             completion(nil, [NSError error:[NSError MHVUnknownError] withDescription:@"The GetApplicationSettings response is invalid."]);
+             
+             return;
+         }
+         
+         completion(applicationSettings, nil);
+     }];
 }
 
-- (void)setApplicationSettingsWithRequestParameters:(NSString *)requestParameters
-                                         completion:(void(^_Nullable)(NSError *_Nullable error))completion
+- (void)setApplicationSettings:(MHVApplicationSettings *)applicationSettings
```

Comment:
I think you should just leave this parameter as "requestParameters". The current code forces the caller to create a string and create an MHVApplicationSettings object - It's just more work for the consumer. It's fine to use the MHVApplicationSettings internally.